frontend/src/herd_pedigree.tsx

```diff
@@ -29,12 +31,12 @@ export function HerdPedigree({ id }: { id: string }) {
   ]
 
   React.useEffect(() => {
-    let mounted = true; // Indicate the mount state
+    const pedigree = herdPedigree(genebanks, id, 5);
+    setPedigree(pedigree)
+  }, [genebanks, id])
```

Comment:
I think you can instead of putting pedigree in React.useState I think you can do 
```typescript
const pedigree = React.useMemo(() => herdPedigree(genebanks, id, 5), [genebanks, id])
```

Reply:
yes! way neater!